src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -20,12 +21,12 @@ public class EnumConverter : TypeConverter
         /// Initializes a new instance of the <see cref='System.ComponentModel.EnumConverter'/> class for the given
         /// type.
         /// </summary>
-        public EnumConverter([DynamicallyAccessedMembers(TypeDescriptor.ReflectTypesDynamicallyAccessedMembers)] Type type)
+        public EnumConverter(Type type)
         {
+            Debug.Assert(type.IsEnum || type.Equals(typeof(Enum)), "type should be an Enum type");
```

Comment:
This is public API - need argument validation not assertion. The code in the instance method seems to assume .IsEnum

Reply:
Now throwing